tracker/derived.go

```diff
@@ -115,24 +116,50 @@ func (m *Model) PatternUnique(t, p int) bool {
 // public getters with further model information
 
 func (m *Model) TracksWithSameInstrumentAsCurrent() []int {
-	currentTrack := m.d.Cursor.Track
-	if currentTrack > len(m.derived.forTrack) {
+	d, ok := m.currentDerivedForTrack()
+	if !ok {
 		return nil
 	}
-	return m.derived.forTrack[currentTrack].tracksWithSameInstrument
+	return d.tracksWithSameInstrument
 }
 
 func (m *Model) CountNextTracksForCurrentInstrument() int {
 	currentTrack := m.d.Cursor.Track
 	count := 0
-	for t := range m.TracksWithSameInstrumentAsCurrent() {
+	for _, t := range m.TracksWithSameInstrumentAsCurrent() {
 		if t > currentTrack {
 			count++
 		}
 	}
 	return count
 }
 
+func (m *Model) CanUseTrackForMidiVelInput(trackIndex int) bool {
+	// makes no sense to record velocity into tracks where notes get recorded
+	tracksForMidiNoteInput := m.TracksWithSameInstrumentAsCurrent()
+	return !slices.Contains(tracksForMidiNoteInput, trackIndex)
+}
+
+func (m *Model) CurrentPlayerConstraints() PlayerProcessConstraints {
+	d, ok := m.currentDerivedForTrack()
+	if !ok {
+		return PlayerProcessConstraints{IsConstrained: false}
+	}
+	return PlayerProcessConstraints{
+		IsConstrained:   m.trackMidiIn,
+		MaxPolyphony:    len(d.tracksWithSameInstrument),
+		InstrumentIndex: d.instrumentRange[0],
+	}
+}
```

Comment:
PlayerProcessConstraints are a new concept - depending on what some settings in the Model are, the Player, via the PlayerProcessContext, might need to take care of some constraining conditions